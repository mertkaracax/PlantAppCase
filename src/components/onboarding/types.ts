import { ReactNode } from "react";
import { DimensionValue } from "react-native";

export type HeaderProps = {
  title: ReactNode;
  subtitle?: string | ReactNode;
  width?: number | "auto" | DimensionValue;
  height?: number | "auto" | DimensionValue;
  top?: number | "auto" | DimensionValue;
  left?: number | "auto" | DimensionValue;
};
