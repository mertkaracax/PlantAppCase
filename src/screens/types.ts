export interface RouteProps {
  navigation?: any;
  route?: any;
}

export enum NavigatorNames {
  OnboardingNavigator = "Onboarding",
  BottomTabNavigator = "BottomTab",
}

export enum OnboardingScreenNames {
  ONBOARDING_ONE = "OnboardingOne",
  ONBOARDING_TWO = "OnboardingTwo",
  ONBOARDING_THREE = "OnboardingThree",
}
