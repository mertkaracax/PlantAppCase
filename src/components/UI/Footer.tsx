import { StyleSheet, Text, View } from "react-native";
import React from "react";
import { ch, cw } from "@src/style/dimensions";
import { Font, FontSize } from "@src/style/fonts";
import { FontStyles } from "@src/style/general";
import { Color } from "@src/style/colors";

const Footer = () => {
  return (
    <Text style={[styles.footer]}>
      By tapping next, you are agreeing to PlantID{" "}
      <Text style={{ textDecorationLine: "underline" }}>Terms of Use</Text> &{" "}
      <Text style={{ textDecorationLine: "underline" }}>Privacy Policy</Text>.
    </Text>
  );
};

export default Footer;

const styles = StyleSheet.create({
  footer: {
    position: "absolute",
    top: ch(740),
    color: Color.FooterText,
    fontSize: FontSize.SIZE11,
    fontFamily: Font.Regular,
    width: cw(232),
    alignSelf: "center",
    textAlign: "center",
  },
});
