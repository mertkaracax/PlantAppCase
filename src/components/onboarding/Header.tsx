import { StyleSheet, Text, View } from "react-native";
import React from "react";
import { HeaderProps } from "./types";
import { ch, cw } from "@src/style/dimensions";
import { FontSize } from "@src/style/fonts";
import { Color } from "@src/style/colors";
import { FontStyles } from "@src/style/general";

const Header: React.FC<HeaderProps> = ({
  title,
  subtitle,
  width = "auto",
  height = "auto",
}) => {
  return (
    <View style={[styles.headerContainer, { width: width, height: height }]}>
      {title}
      {subtitle && (
        <Text style={[styles.subtitle, FontStyles.rubik]}>{subtitle}</Text>
      )}
    </View>
  );
};

export default Header;

const styles = StyleSheet.create({
  headerContainer: {
    marginLeft: cw(24),
    marginTop: ch(12),
    padding: 8,
    lineHeight: 33.3,
  },
  title: {
    fontWeight: "300",
    fontSize: FontSize.SIZE28,
    color: Color.HeaderText,
  },
  subtitle: {
    color: Color.SubheaderText,
    fontWeight: "300",
    fontSize: FontSize.SIZE16,
  },
});
