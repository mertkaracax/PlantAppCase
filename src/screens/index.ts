import OnboardingOne from "../screens/onboarding/OnboardingOne";
import OnboardingTwo from "./onboarding/OnboardingTwo";
import OnboardingThree from "./onboarding/OnboardingThree";
import Home from "./home/Home";

export { Home, OnboardingOne, OnboardingTwo, OnboardingThree };
