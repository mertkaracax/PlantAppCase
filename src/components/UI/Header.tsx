import { StyleSheet, Text, View } from "react-native";
import React, { memo } from "react";
import { HeaderProps } from "../onboarding/types";
import { ch, cw } from "@src/style/dimensions";
import { FontStyles } from "@src/style/general";

const Header: React.FC<HeaderProps> = ({
  title,
  subtitle,
  width = cw(315),
  height = "auto",
  top,
  left,
  marginTop = ch(12),
  marginBottom,
  maxWidth,
  children,
}) => {
  return (
    <View
      style={[
        styles.headerContainer,
        {
          width: width,
          maxWidth: maxWidth,
          height: height,
          top: top,
          left: left,
          marginTop: marginTop,
          marginBottom: marginBottom,
        },
      ]}
    >
      {title}
      {typeof subtitle === "string" ? (
        <Text style={[FontStyles.headerSubtitle]}>{subtitle}</Text>
      ) : (
        React.isValidElement(subtitle) && subtitle
      )}
      {children}
    </View>
  );
};

export default memo(Header);

const styles = StyleSheet.create({
  headerContainer: {
    marginLeft: cw(24),
    marginTop: ch(12),
  },
});
