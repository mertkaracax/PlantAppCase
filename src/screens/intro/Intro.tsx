import { Text, StyleSheet, Image, View, ImageBackground } from "react-native";
import React from "react";
import { SafeAreaView } from "react-native-safe-area-context";
import { RouteProps } from "../types";
import { ch, cw } from "@src/style/dimensions";
import Header from "@src/components/UI/Header";
import { Font } from "@src/style/fonts";
import { IMAGES } from "@src/assets/images";
import Footer from "@src/components/UI/Footer";
import OnboardingButton from "@src/components/onboarding/OnboardingButton";
import { FontStyles } from "@src/style/general";

const Intro: React.FC<RouteProps> = ({ route, navigation }) => {
  const headerTitle = (
    <Text style={[FontStyles.headerTitle, { maxWidth: cw(375) }]}>
      Welcome to <Text style={{ fontFamily: Font.SemiBold }}>PlantApp</Text>
    </Text>
  );

  return (
    <ImageBackground
      style={{ flex: 1 }}
      source={require("@src/assets/images/onboarding/Onboarding_1_white_bg.png")}
    >
      <SafeAreaView style={styles.root}>
        <Header
          title={headerTitle}
          subtitle={`Identify more than 3000+ plants and\n 88% accuracy.`}
          height={ch(85)}
          marginBottom={ch(24)}
        />
        <View style={styles.imageContainer}>
          <Image style={styles.image} source={IMAGES.onboarding_1.background} />
        </View>
        <OnboardingButton
          style={{ height: ch(56) }}
          title="Get Started"
          onPress={() => {
            navigation.navigate("Onboarding");
          }}
        />
        <Footer />
      </SafeAreaView>
    </ImageBackground>
  );
};

const styles = StyleSheet.create({
  root: {
    flex: 1,
  },
  imageContainer: {
    height: ch(499),
    width: "100%",
  },
  image: {
    resizeMode: "contain",
    alignSelf: "center",
    flex: 1,
  },
});

export default Intro;
