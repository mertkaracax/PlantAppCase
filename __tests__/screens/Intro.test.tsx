import React from "react";
import Intro from "@src/screens/intro/Intro";
import renderer, {
  ReactTestRenderer,
  ReactTestInstance,
} from "react-test-renderer";
import { RouteProps } from "@src/screens/types";
import Header from "@src/components/UI/Header";
import Footer from "@src/components/UI/Footer";

jest.mock("@src/components/UI/Header", () => "Header");
jest.mock(
  "@src/components/onboarding/OnboardingButton",
  () => "OnboardingButton"
);
jest.mock("@src/components/UI/Footer", () => "Footer");
jest.mock("react-native-safe-area-context", () => ({
  SafeAreaView: "SafeAreaView",
}));
jest.mock("@src/assets/images", () => ({
  IMAGES: {
    onboarding_1: {
      background: "background.png",
    },
  },
}));

describe("Intro", () => {
  let component: ReactTestRenderer;
  let root: ReactTestInstance;
  const mockNavigate = jest.fn();

  beforeEach(() => {
    component = renderer.create(
      <Intro navigation={{ navigate: mockNavigate }} route={{} as RouteProps} />
    );
    root = component.root;
  });

  test("renders correctly", () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("navigates to Onboarding when Get Started button is pressed", () => {
    const button = root.findByProps({ title: "Get Started" });
    button.props.onPress();
    expect(mockNavigate).toHaveBeenCalledWith("Onboarding");
  });

  test("renders Footer component", () => {
    const footer = root.findByType(Footer);
    expect(footer).toBeTruthy();
  });
});
