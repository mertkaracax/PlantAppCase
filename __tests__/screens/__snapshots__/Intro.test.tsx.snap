// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Intro renders correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    {
      "flex": 1,
    }
  }
>
  <Image
    source={1}
    style={
      [
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <SafeAreaView
    style={
      {
        "flex": 1,
      }
    }
  >
    <Header
      height={139.64285714285714}
      marginBottom={39.42857142857142}
      subtitle="Identify more than 3000+ plants and
 88% accuracy."
      title={
        <Text
          style={
            [
              {
                "color": "#13231B",
                "fontFamily": "Rubik_400Regular",
                "fontSize": 56.00000000000001,
                "letterSpacing": 0.14,
                "lineHeight": 33.18,
                "maxWidth": 630,
              },
              {
                "maxWidth": 750,
              },
            ]
          }
        >
          Welcome to 
          <Text
            style={
              {
                "fontFamily": "Rubik_600SemiBold",
              }
            }
          >
            PlantApp
          </Text>
        </Text>
      }
    />
    <View
      style={
        {
          "height": 819.7857142857143,
          "width": "100%",
        }
      }
    >
      <Image
        source="background.png"
        style={
          {
            "alignSelf": "center",
            "flex": 1,
            "resizeMode": "contain",
          }
        }
      />
    </View>
    <OnboardingButton
      onPress={[Function]}
      style={
        {
          "height": 92,
        }
      }
      title="Get Started"
    />
    <Footer />
  </SafeAreaView>
</View>
`;
